guides/dataloader/testing.md

```diff
@@ -60,10 +79,10 @@ def test_it_fetches_objects_by_id
       req4 = dataloader.with(Sources::ActiveRecord).request(-1)
 
       # Validate source's matching up of records
-      expect(req1.load).to eq(user_1)
-      expect(req2.load).to eq(user_3)
-      expect(req3.load).to eq(user_2)
-      expect(req4.load).to be_nil
+      expect(req1).to eq(user_1)
+      expect(req2).to eq(user_2)
+      expect(req3).to eq(user_3)
+      expect(req4).to be_nil
```

Comment:
I don't think this change is right, for a couple reasons: 

- the examples above use `.request`, which registers the data dependency with `Dataloader` but _doesn't_ yield the fiber; it returns a pending request which must be `.load`-ed in order to return the actual value. I think we need the `.load` calls here. 
- In the example above, `req2` is a request for `user_3.id`; I would expect `req2.load` to return `user_3`, not `user_2`.

If you think that this will work without `.load`, would you mind sharing a small replication script where `.request` returns the dataloaded object?

Reply:
the example above failed my tests. Here's roughly what the source looks like: 

```rb
class Sources::User < GraphQL::Dataloader::Source
  def initialize(model_class) # rubocop:disable Lint/MissingSuper
    @model_class = model_class
  end

   def fetch(ids)
    records = @model_class.where(id: ids)

    # Return a list of records, with `nil` for any ID that wasn't found
    ids.map do |uuid|
      # Return the record if there's a match
      records.find do |record|
        record.id == id
      end
    end
  end
end
```

Reply:
Ah i was doing `.load(id)` instead of `.request(id)`